nixos/doc/manual/development/modular-services.md

```diff
@@ -0,0 +1,91 @@
+
+# Modular Services {#modular-services}
+
+Status: in development. This functionality is new in NixOS 25.05, and significant changes should be expected. We'd love to hear your feedback in <!-- FIXME PR link -->
+
+Traditionally, NixOS services were defined using sets of options *in* modules, not *as* modules. This made them non-modular, resulting in problems with composability, reuse, and portability.
+
+A *modular service* is a [module] that defines values for a core set of options, including which program to run.
```

Comment:
I'm trying to understand what this text is trying to tell me, and it seems we're missing some prerequisites. Please correct this if I'm getting it wrong.

```suggestion
A configuration management framework is an application of `evalModules` with the `class` and `specialArgs` input attribute set to particular values.
NixOS is such a configuration management framework, and so are [Home Manager](https://github.com/nix-community/home-manager) and [nix-darwin](https://github.com/lnl7/nix-darwin).

A *modular service* is a [module] that defines values for a core set of options declared by the service management component of a configuration management framework , including which program to run.
```